Library/Homebrew/utils/github.rb

```diff
@@ -228,34 +228,15 @@ def raise_api_error(output, errors, http_code, headers, scopes)
   end
 
   def issues_matching(query, qualifiers = {})
-    uri = URI.parse("#{API_URL}/search/issues")
-    uri.query = build_query_string(query, qualifiers)
-    open(uri) { |json| json["items"] }
+    search("issues", query, **qualifiers)
   end
 
   def repository(user, repo)
-    open(URI.parse("#{API_URL}/repos/#{user}/#{repo}"))
+    open(url_to("repos", user, repo))
   end
 
-  def search_code(*params)
-    uri = URI.parse("#{API_URL}/search/code")
-    uri.query = "q=#{uri_escape(params.join(" "))}"
-    open(uri) { |json| json["items"] }
-  end
-
-  def build_query_string(query, qualifiers)
-    s = "q=#{uri_escape(query)}+"
-    s << build_search_qualifier_string(qualifiers)
-    s << "&per_page=100"
-  end
-
-  def build_search_qualifier_string(qualifiers)
-    {
-      repo: "Homebrew/homebrew-core",
-      in: "title",
-    }.update(qualifiers).map do |qualifier, value|
-      "#{qualifier}:#{value}"
-    end.join("+")
+  def search_code(**params)
+    search("code", **params)
   end
 
   def uri_escape(query)
```

Comment:
Maybe just replace `uri_escape` with `URI.encode_www_form_component`, since we're only using it once.